WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -108,34 +123,36 @@ protected void onStart() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && source == Source.STATS) {
-            showStats();
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResultAndFinish();
         } else {
-            finish();
+            finishAndGoBackToSource();
         }
     }
 
-    private void showStats() {
-        if (!TextUtils.isEmpty(reason)) {
-            ToastUtils.showToast(this, reason);
-            finish();
-        } else {
-            SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
-            mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(site));
-        }
+    private void reloadSite() {
+        //Reloads the site
+        SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
+        mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(site));
     }
 
     @Override
     public void onBackPressed() {
         super.onBackPressed();
-        finish();
+        finishAndGoBackToSource();
     }
 
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(SiteStore.OnSiteChanged event) {
```

Comment:
We really shouldn't do this. I know it's not really in the changeset, however a site fetch might be started from any other place and this will just close the activity even though it's not what we wanted to do.